Github.js

```diff
@@ -0,0 +1,421 @@
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {	
+	var item = new Z.Item("computerProgram");
+	
+	//basic metadata from the meta tags in the head
+	item.url = ZU.xpathText(doc, '/html/head/meta[@property="og:url"]/@content');
+	item.title = ZU.xpathText(doc, '/html/head/meta[@property="og:title"]/@content');
+	item.abstractNote = ZU.xpathText(doc, '/html/head/meta[@property="og:description"]/@content');
+	item.libraryCatalog = "GitHub";
+	item.rights = ZU.xpathText(doc, '//a[*[contains(@class, "octicon-law")]]');
+	
+	//api calls to /repos and /repos/../../stats/contribuors
+	var apiUrl = "https://api.github.com/repos/"+item.title;
+	ZU.doGet(apiUrl, function(result) {
+		var json = JSON.parse(result);
+		//Z.debug(json);
+		item.programmingLanguage = json.language;
+		
+		item.extra = "original-date: " + json.created_at;
+		item.date = json.updated_at;
+		ZU.doGet(apiUrl+"/stats/contributors", function(contributors) {
+			var jsonContributors = JSON.parse(contributors);
+			//it seems that the contributors with most contributions are
+			//at the end of this list --> loop trough in inverse direction
+			for (var i=jsonContributors.length-1; i>-1; i--) {
+				var contributor = jsonContributors[i].author;
+				item.creators.push({ "lastName": contributor.login, creatorType : 'contributor', fieldMode : 1 });
+				//getAuthor(contributor.login);
+			}
+			item.attachments.push({
+				title: "Snapshot",
+				document: doc
+			});
+			
+			item.complete();
+		});
+
+	});
+	
+
+}
+
+
+// get the full name from the author profile page
+function getAuthor(username) {
+	var url = "https://github.com/" + encodeURIComponent(username);	
+	ZU.processDocuments(url, function(text) {
+		var author = ZU.xpathText(text, '//span[contains(@class, "vcard-fullname")]');
+		if (!author) { author = ZU.xpathText(text, '//span[contains(@class, "vcard-username")]'); }
+		if (!author) { author = ZU.xpathText(text, '/html/head/meta[@property="profile:username"]/@content'); }
+		Z.debug(author);
+		author = ZU.cleanAuthor(author, "author");
+	});
+	// temporary, until we get the author string out of the closure
+	return ZU.cleanAuthor(username, "author");
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://github.com/zotero/zotero/",
+		"items": [
+			{
+				"itemType": "computerProgram",
+				"title": "zotero/zotero",
+				"creators": [
+					{
+						"lastName": "dstillman",
```

Comment:
these giant contributor lists -- for CSL these would be many hundred people in the styles directory -- don't make much sense to me. I'd tend to say we don't want them. What's the argument for? And wouldn't it make sense to list the account or the organization of the repo as the author?